metaflow/plugins/kfp/kfp.py

```diff
@@ -28,35 +29,166 @@ def get_ordered_steps(graph):
     return ordered_steps
 
 
-def step_container_op(step_name, code_url=DEFAULT_FLOW_CODE_URL):
+def step_op_func(step_name: str,
+                 code_url: str,
+                 ds_root: str,
+                 run_id: str,
+                 task_id: str,
+                 prev_step_name: str,
+                 prev_task_id: str) -> NamedTuple('StepOutput', [('ds_root', str),
+                                                         ('run_id', str),
+                                                         ('next_step', str),
+                                                         ('next_task_id', str),
+                                                         ('current_step', str),
+                                                         ('current_task_id', str)]
+                                                      ):
     """
-    Method to create a kfp container op that corresponds to a single step in the flow.
+    Function used to create a KFP container op (see: `step_container_op`) that corresponds to a single step in the flow.
 
-    TODO: This does not maintain state. The custom pre-start command used below would be removed once we have state accessible across KFP steps.
-    TODO: The public docker is a copy of the internal docker image we were using (borrowed from aip-kfp-example). Check if any stage here may need to be further modified later.
     """
+    import subprocess
+    from collections import namedtuple
+
+    print("\n----------RUNNING: CODE DOWNLOAD from URL---------")
+    subprocess.call(["curl -o helloworld.py {}".format(code_url)], shell=True)
+
+    print("\n----------RUNNING: KFP Installation---------------")
+    subprocess.call(["pip3 install kfp"], shell=True) # TODO: Remove this once KFP is added to dependencies
+
+    print("\n----------RUNNING: METAFLOW INSTALLATION----------")
+    subprocess.call(["pip3 install --user --upgrade git+https://github.com/zillow/metaflow.git@state-integ-s3"],
+                    shell=True)
+
+    print("\n----------RUNNING: MAIN STEP COMMAND--------------")
+    S3_BUCKET = "s3://workspace-zillow-analytics-stage/aip/metaflow"
+    S3_AWS_ARN = "arn:aws:iam::170606514770:role/dev-zestimate-role"
```

Comment:
S3_BUCKET and S3_BUCKET should be env variables and not hard coded here:
```
METAFLOW_DATASTORE_SYSROOT_S3
METAFLOW_AWS_ARN
```

Reply:
Done. I followed [this document](https://www.kubeflow.org/docs/pipelines/sdk/enviroment_variables/). Could you confirm if this is what you have in mind?